src/components/PolarisTestProvider/PolarisTestProvider.tsx

```diff
@@ -82,7 +83,11 @@ export function PolarisTestProvider({
   // I'm not that worried about it
   const appBridgeApp = appBridge as React.ContextType<typeof AppBridgeContext>;
 
-  const mergedTheme = buildThemeContext(theme);
+  const {unstableGlobalTheming = false} = features;
+  const customProperties = unstableGlobalTheming
+    ? buildCustomProperties(theme, unstableGlobalTheming)
+    : undefined;
+  const mergedTheme = buildThemeContext(theme, customProperties);
```

Comment:
This mimics the theme provider by putting the new CSS customer properties in the test environment context, but not the legacy properties.